### PR TITLE
Extend EntryExpiredListener for EntryListener interface

### PR DESCRIPTION
**Pull request by ahmetmircik**
closes https://github.com/hazelcast/hazelcast/issues/14958

### CODE REVIEW

**Comment by ahmetmircik** [comment]
run-lab-run

**Comment by mmedenjak** [comment]
@ahmetmircik can you rebase and see if any comments need to be addressed?

File: hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/replicatedmap/AbstractReplicatedMapAddEntryListenerMessageTask.java [hunk 1]
```diff
@@ -152,6 +152,11 @@ public void entryEvicted(EntryEvent<Object, Object> event) {
         handleEvent(event);
     }
 
+    @Override
+    public void entryExpired(EntryEvent<Object, Object> event) {
+        // TODO handle this event
```
[the remark this answers is not in the document]
Comment (ahmetmircik) [reply hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/replicatedmap/AbstractReplicatedMapAddEntryListenerMessageTask.java]: fixed

File: hazelcast/src/test/java/com/hazelcast/multimap/MultiMapSplitBrainTest.java [hunk 2]
```diff
@@ -473,34 +474,14 @@ private static void assertMultiMapsSize(MultiMap<?, ?> multiMap1, MultiMap<?, ?>
         assertEqualsStringFormat("backupMultiMap should have size %d, but was %d", expectedSize, actualBackupSize);
     }
 
-    private static class EmptyEntryListener<K, V> implements EntryListener<K, V> {
+    private static class EmptyEntryListener<K, V> extends EntryAdapter<K, V> {
         @Override
-        public void mapEvicted(MapEvent event) {
+        public void onEntryEvent(EntryEvent<K, V> event) {
```
[the remark this answers is not in the document]
Comment (ahmetmircik) [reply hazelcast/src/test/java/com/hazelcast/multimap/MultiMapSplitBrainTest.java]: fixed

File: hazelcast/src/test/java/com/hazelcast/multimap/MultiMapSplitBrainTest.java [hunk 3]
```diff
@@ -473,34 +474,14 @@ private static void assertMultiMapsSize(MultiMap<?, ?> multiMap1, MultiMap<?, ?>
...
-        }
-
-        @Override
-        public void entryEvicted(EntryEvent<K, V> event) {
-
-        }
-
-        @Override
-        public void entryAdded(EntryEvent<K, V> event) {
+        public void onMapEvent(MapEvent event) {
```
[the remark this answers is not in the document]
Comment (ahmetmircik) [reply hazelcast/src/test/java/com/hazelcast/multimap/MultiMapSplitBrainTest.java]: fixed

**Comment by ahmetmircik** [comment]
@mmedenjak rebased.

**Comment by joschi** [comment]
We just chased and could fix a bug in our system after upgrading to Hazelcast 4.0.2 because `EntryEvictedListener` wasn't called anymore when entries expired.

Would it be possible to make it clearer in the release notes for Hazelcast 4.0 that `EntryEvictedListener` isn't necessarily called anymore and that you have to add an `EntryExpiredListener` if you want to handle expiration events?

The release notes at https://docs.hazelcast.org/docs/rn/index.html#distributed-events currently say:
> Added `EntryExpiredListener` to the `EntryListener` interface. #14959

Refs https://github.com/hazelcast/hazelcast/issues/14712

**Comment by mmedenjak** [comment]
Thanks for the report @joschi . @Serdaro can you add this? ☝️

**Pull request merged**